common/src/main/java/com/wynntils/services/mapdata/attributes/DerivedAttributes.java

```diff
@@ -8,50 +8,51 @@
 import com.wynntils.services.mapdata.attributes.type.MapDecoration;
 import com.wynntils.utils.colors.CustomColor;
 import com.wynntils.utils.render.type.TextShadow;
+import java.util.Optional;
 import java.util.function.Function;
 
 public abstract class DerivedAttributes implements MapAttributes {
     protected abstract <T> T getAttribute(Function<MapAttributes, T> getter);
 
     @Override
-    public String getLabel() {
+    public Optional<String> getLabel() {
         return getAttribute(MapAttributes::getLabel);
     }
 
     @Override
-    public String getIconId() {
+    public Optional<String> getIconId() {
         return getAttribute(MapAttributes::getIconId);
     }
 
     @Override
-    public int getPriority() {
-        Integer integer = getAttribute(MapAttributes::getPriority);
-        return integer == null ? 0 : integer;
+    public Optional<Integer> getPriority() {
+        Optional<Integer> integer = getAttribute(MapAttributes::getPriority);
+        return integer.isEmpty() ? Optional.of(0) : integer;
```

Comment:
Should we really do this, or just return ` getAttribute(MapAttributes::getPriority)`? (The same question applies to `getLevel` below.)

Judging from symmetry reason with the other methods, this looks a bit sus.

Reply:
You already commented about not having a default priority. This should be set to it. This 0 is basically the fallback value, which is correct for level, but not correct here. I'll set the default 500 here.